application_test.go

```diff
@@ -427,3 +427,20 @@ func TestApplicationConfiguration(t *testing.T) {
 	assert.True(t, ok)
 	assert.Equal(t, ErrCodeNotFound, apiErr.ErrCode)
 }
+
+func TestWaitOnApplication(t *testing.T) {
+	endpoint := newFakeMarathonEndpoint(t, nil)
+	defer endpoint.Close()
+	client := endpoint.Client.(*marathonClient)
+
+	c := make(chan bool)
+	err := client.waitOnApplication(fakeAppName, c)
+	assert.NoError(t, err)
+
+	c = make(chan bool)
+	go func() {
+		c <- true
+	}()
+	err = client.waitOnApplication("no_such_app", c)
+	assert.IsType(t, err, ErrTimeoutError)
```

Comment:
`assert.EqualError` works with strings, but I can change assertion to
 `assert.EqualError(t, err, ErrTimeoutError.String())` if you think it will be clearer